File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] answer = {0,0};
        for(int i = 0; i < nums.length; i++)
        {
            int diff = target - nums[i];

            if(map.containsKey(diff))
            {
                answer[0] = map.get(diff);
                answer[1] = i;
                return answer;
            }
            else
            {
                map.put(nums[i], i);
            }
        }

        return answer;
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        f, l = 0, len(nums)

        for i in range(len(nums)):
            half = (f + l) // 2

            print(f"{f} {l}")

            if nums[half] == target: return half

            if target < nums[half]:
                l = half
            else:
                f = half
        
        return -1
        